### PR TITLE
Upgrade TVM to 0.10.0

### PR DESCRIPTION
**Pull request by lucifer1004**

### CODE REVIEW

File: T/TVM/build_tarballs.jl [hunk 1]
```diff
@@ -1,69 +1,76 @@
...
-    LibraryProduct("libnnvm_compiler", :libnnvm_compiler),
 ]
 
 # Dependencies that must be installed before this package can be built
-dependencies = [
-    "Zlib_jll",
-    "LLVM_jll",
+dependencies = Dependency[
+    Dependency(PackageSpec(name="Zlib_jll")),
+    Dependency(PackageSpec(name="LLVM_jll")),
```
Comment (giordano) [inline T/TVM/build_tarballs.jl:71]: Do you need full llvm or just libllvm (which is a much smaller package)?  Also, if this links dynamically to libllvm at runtime, you're locking into a single llvm.  You want to see what https://github.com/JuliaPackaging/Yggdrasil/blob/79007df5f1de4a7b58119e87b5a7ad0ee6e984f0/L/LLVMExtra/build_tarballs.jl does.

**Comment by lucifer1004** [comment]
I then switched to GCC 9, and the error became ` collect2: fatal error: cannot find ‘ld’` (see https://buildkite.com/julialang/yggdrasil/builds/536#01860cec-9b1d-4539-8328-e44e8347d692)

**Comment by lucifer1004** [comment]
Static linking is more difficult than I thought.

**Comment by lucifer1004** [comment]
Seems to be working.

**Comment by imciner2** [comment]
Closing due to inactivity. Please submit a new PR on top of the current master branch if you still want to include this recipe.

**Pull request closed without merging**